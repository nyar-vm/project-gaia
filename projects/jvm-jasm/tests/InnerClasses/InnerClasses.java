public class InnerClasses {
    private String outerField = "Outer";
    
    public class Inner {
        private String innerField = "Inner";
        
        public String getOuterField() {
            return outerField;
        }
    }
    
    public static class StaticInner {
        private static int staticCounter = 0;
        
        public static void increment() {
            staticCounter++;
        }
    }
    
    public void localInner() {
        final String localVar = "Local";
        
        class LocalInner {
            public String getLocal() {
                return localVar;
            }
        }
        
        LocalInner local = new LocalInner();
    }
    
    public void anonymousInner() {
        Runnable runnable = new Runnable() {
            @Override
            public void run() {
                System.out.println("Anonymous");
            }
        };
    }
}