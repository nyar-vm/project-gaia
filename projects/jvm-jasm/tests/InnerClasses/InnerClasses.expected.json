{
    "class_name": "InnerClasses",
    "class_modifiers": [
        "public",
        "super"
    ],
    "method_count": 3,
    "field_count": 0,
    "expected_methods": [
        "<init>:()V",
        "localInner:()V",
        "anonymousInner:()V"
    ],
    "expected_fields": [],
    "syntactic_sugar_features": [],
    "check_tokens": true,
    "source_file": "InnerClasses.java"
}